hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/utils/FileServiceUtils.java

```diff
@@ -124,6 +125,36 @@ public static void validateAndAddRequiredKeys(
         }
     }
 
+    /**
+     * The function is unique for File delete Pre-handle validates the keys.
+     * Then it wraps the file Keylist into thresholds adds them to the context.
+     * In addition, it add the transaction keys to the context.
+     *
+     * @param file file that will be checked for required keys
+     * @param transactionKeys transaction keys that add to context for required keys.
+     * @param context the prehandle context for the transaction.
+     */
+    public static void validateAndAddRequiredKeysForDelete(
+            @Nullable final File file,
+            @Nullable final KeyList transactionKeys,
+            @NonNull final PreHandleContext context) {
+        if (file != null) {
+            KeyList fileKeyList = file.keys();
+
+            if (fileKeyList != null && fileKeyList.hasKeys()) {
+                ThresholdKey syntheticKey =
+                        ThresholdKey.newBuilder().threshold(1).keys(fileKeyList).build();
+                context.requireKey(Key.newBuilder().thresholdKey(syntheticKey).build());
+            }
+        }
+
+        if (transactionKeys != null && transactionKeys.hasKeys()) {
```

Comment:
What's transaction keys?

Reply:
These are the keys in context.body().keys() and need to be added as well.

Reply:
context.body(): TransactionBody, I don't see any keys in TransactionBody.

Reply:
you are right for delete not exist we send null I will remove and fix it right away. thank you great catch